rclcpp/include/rclcpp/node_interfaces/node_logger.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__NODE_INTERFACES__NODE_LOGGER_HPP_
+#define RCLCPP__NODE_INTERFACES__NODE_LOGGER_HPP_
+
+#include <memory>
+
+#include "rclcpp/logger.hpp"
+#include "rclcpp/macros.hpp"
+#include "rclcpp/node_interfaces/node_base_interface.hpp"
+#include "rclcpp/node_interfaces/node_logger_interface.hpp"
+#include "rclcpp/visibility_control.hpp"
+
+namespace rclcpp
+{
+namespace node_interfaces
+{
+
+/// Implementation of the NodeLogger part of the Node API.
+class NodeLogger : public NodeLoggerInterface
```

Comment:
I have a small concern here that `NodeLogger` sounds to much like a thing that logs for a node (which is not really what it is). What about `NodeLogging` and `NodeLoggingInterface` instead? I had a similar dilemma with `NodeTopic` versus `NodeTopics`, but I ended up going with `NodeTopics` because `NodeTopicsInterface` sounded better. For the same logic I think decided to suggest `NodeLogging` here, but I'll leave it to your discretion.

Reply:
Yeah, the `NodeLogger` attribute of the `Node`/`LifecycleNode` class is not a logger but a way to access the logger. will switch to `NodeLogging` as proposed